src/instana/agent/host.py

```diff
@@ -297,73 +324,147 @@ def report_data_payload(self, payload):
         except urllib3.exceptions.MaxRetryError:
             pass
         except Exception as exc:
-            logger.debug("report_data_payload: Instana host agent connection error (%s)", type(exc), exc_info=True)
+            logger.debug(
+                f"report_data_payload: Instana host agent connection error ({type(exc)})",
+                exc_info=True,
+            )
         return response
 
-    def handle_agent_tasks(self, task):
+    def report_metrics(self, payload: Dict[str, Any]) -> Union[Response, None]:
+        metric_count = len(payload["metrics"])
```

Comment:
It would be better to avoid `KeyError` exceptions here, even if `metrics` is a key that always will be available. I suggest using the `get()` method since this will always be a dictionary.

```python
>>> payload = {"spans": ["ha", "he"]}
>>> metric_count = len(payload["metrics"])
Traceback (most recent call last):
  File "<python-input-7>", line 1, in <module>
    metric_count = len(payload["metrics"])
                       ~~~~~~~^^^^^^^^^^^
KeyError: 'metrics'
>>> metric_count = len(payload.get("metrics", []))
>>> metric_count
0
>>>
```

 ```suggestion
        metric_count = len(payload.get("metrics", []))
```